### PR TITLE
[#176536899] Test credit card submission form

### PR DESCRIPTION
**Pull request by GiovanniMancini**
This PR proposes to add a set of tests to guarantee that credit cards are properly handled by the submission form. In particular, I've added tests for each circuit supported by the Payment Manager. Tests are executed in a browser environment controlled via `puppeteer`, they are based on files served by an `express` application with a `static` middleware, which emulates the server spawned by `yarn start`. Tests try to emulate user behavior when she inserts credit card PAN. Asserts try to catch wrong credit cards from visual appearance of the page or state of DOM elements.  

#### List of Changes
- Dev server which serves the frontend as bundled by `parcel`
- Test suite for credit card numbers

#### Motivation and Context
Proposed tests guarantee that user receives a feedback as soon as possible when she enters a wrong credit card number or an unsupported card.

#### How Has This Been Tested?
Integration tests in browser environment + local dev server

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/__integrations__/creditCard.devServer.integration.test.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    const creditCardFieldS = '#creditcardnumber';
+
+    await page.focus(creditCardFieldS);
+    await page.keyboard.type('4111111111111111');
+    const checked = await page.$eval(creditCardFieldS, element => element.getAttribute('data-checked'));
+    expect(checked).toMatch(/1/);
+    await page.close();
+  });
+
+  it('should handle Maestro cards', async () => {
```
Comment (infantesimone) [inline src/__integrations__/creditCard.devServer.integration.test.ts:89]: ditto

File: src/__integrations__/creditCard.devServer.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    await page.goto(`http://${HOST}:${PORT}/index.html?p=1234`);
+    await page.setViewport({ width: 1200, height: 907 });
+
+    const creditCardFieldS = '#creditcardnumber';
+    const creditCardInput = await page.$(creditCardFieldS);
+
+    await page.focus(creditCardFieldS);
+    await page.keyboard.type('5555555555554444');
+    const checked = await page.$eval(creditCardFieldS, element => element.getAttribute('data-checked'));
+    expect(checked).toMatch(/1/);
```
Comment (pasqualespica) [inline src/__integrations__/creditCard.devServer.integration.test.ts:139]: why in the previously test do you use `toMatch` and here `resolves`  to check `data-checked` equal to `1` ?

**Review by pasqualespica — APPROVED** [review]
Great work 👍 
I think these tests with `puppeteer` are really a great added value. I also believe an example for our future projects.
It'd be even more useful to have tests to check that the card is truly supported. And don't find yourself a KO during a payment  instead.
It wouldn't be a nice user scenario.

File: src/__integrations__/creditCard.devServer.integration.test.ts (hunk 2 above)
Comment (GiovanniMancini) [reply src/__integrations__/creditCard.devServer.integration.test.ts:139]: @pasqualespica  Thank you for pointing out. `page.$eval()` returns a `Promise`, so if I call it _inside_ `expect`, as done in previous test, I have to use `resolves` to guarantee that it resolves. On the other hand, If I `await` _outside_ expect, as done in this test, it's no longer a `Promise` but a resolved value, so using `toMach()` is sufficient. I just experimented with different ways of asserting on `Promises` but the two tests, as you pointed out, are perfectly equivalent

**Comment by GiovanniMancini** [comment]
@pasqualespica Thank you for your review, for negative tests I think we should wait the validation logic to be implemented on frontend side
> Great work 👍
> I think these tests with `puppeteer` are really a great added value. I also believe an example for our future projects.
> It'd be even more useful to have tests to check that the card is truly supported. And don't find yourself a KO during a payment instead.
> It wouldn't be a nice user scenario.

**Review by infantesimone — APPROVED** [review]

**Pull request merged**